src/vsr/replica.zig

```diff
@@ -5058,21 +5047,15 @@ pub fn ReplicaType(
         }
 
         /// Returns the oldest op that the replica must/(is permitted to) repair.
-        /// The goal is to repair as much as possible without triggering unnecessary state sync.
         ///
-        /// - Repairing uncommitted ops is necessary as primary — we will need them to proceed.
-        /// - Repairing committed + not-yet-checkpointed ops is useful because we might crash,
-        ///   in which case we will need them to recover.
-        /// - Repairing committed + checkpointed ops:
-        ///   - backups don't repair checkpointed ops because:
-        ///     - repairing an op from the previous checkpoint might "clean" a slot which belongs
-        ///       to a corrupt entry from the current WAL-wrap, leading to the backup incorrectly
-        ///       nacking the latter entry.
-        ///     - there is no guarantee that they will ever be available (if our head is behind),
-        ///       and we don't want to stall the new view startup.
-        ///   - primaries do repair checkpointed ops so that they can help lagging backups catch up,
-        ///     as long as the op is new enough to be present in the WAL of some other replica and
-        ///     is from the previous checkpoint.
+        /// Safety condition: repairing an old op must not overwrite a newer op from the next wrap.
+        ///
+        /// Availability condition: each committed op must be present either in a quorum of WALs or
+        /// it in a quorum of checkpoints.
+        ///
+        /// If op=trigger+1 is committed, the corresponding checkpoint is durably present on
```

Comment:
Ah,  but I think this is wrong after https://github.com/tigerbeetle/tigerbeetle/pull/1508!

The condition is now `op=prepare_max + 1`, not `op=trigger + 1` I think!